src/sentry/api/serializers/rest_framework/project_key.py

```diff
@@ -2,13 +2,23 @@
 
 from sentry.api.fields.empty_integer import EmptyIntegerField
 from sentry.loader.browsersdkversion import get_browser_sdk_version_choices
+from sentry.loader.dynamic_sdk_options import DynamicSdkLoaderOption
 
 
 class RateLimitSerializer(serializers.Serializer):
     count = EmptyIntegerField(min_value=0, required=False, allow_null=True)
     window = EmptyIntegerField(min_value=0, max_value=60 * 60 * 24, required=False, allow_null=True)
 
 
+class DynamicSdkLoaderOptionDictField(serializers.DictField):
+    """Validates dynamicSdkLoaderOptions to make sure it only contains valid keys"""
+
+    def to_internal_value(self, data):
+        if set(data.keys()) != {o.value for o in DynamicSdkLoaderOption}:
```

Comment:
This will reject any additional keys, which can make forwards/backwards compatibility harder. Do you need to validate the values of these keys or is any datatype acceptable?

Reply:
Ah great point - we must have forwards compatibility here. Let me rework this, thanks for pointing it out.